daemon/src/wire.rs

```diff
@@ -664,6 +671,44 @@ impl From<CfdTransactions> for RolloverMsg1 {
     }
 }
 
+#[derive(Debug, Copy, Clone, Serialize, Deserialize)]
+pub enum Environment {
```

Comment:
I probably wouldn't bother with the second enum - the risk of someone setting `Legacy` in the envvar is minimal.

Unknown could have been handled by `Option<Environment>`, if it wasn't set it must be unknown (that gets rid of awkwardness when parsing and arguably is more natural, as it's a first-class way to encode that it's unknown).

Reply:
Setting `Legacy` in the envvar will still default to `UNKNOWN`. But that's not the point. We serialize / deserialize over the wire while we perform other operations (e.g. parse) with `daemon::Environment`. I think we should clearly separate `wire` types to avoid breaking them.

Reply:
fair enough!